tests/TestCase/Model/Behavior/ProfferBehaviorTest.php

```diff
@@ -933,4 +947,81 @@ public function testMultipleFieldUpload()
         $this->assertEquals('image_480x640.jpg', $entity->get('avatar'));
         $this->assertEquals('proffer_test', $entity->get('avatar_dir'));
     }
+
+    /**
+     * Test that uploads are processed correctly when the upload is it's own entity. For when users associate many
+     * uploads with a single parent item. Such as Posts hasMany Uploads
+     *
+     * @return void
+     */
+    public function testMultipleAssociatedUploads()
+    {
+        $eventManager = $this->createMock('Cake\Event\EventManager');
+
+        $uploadsSchema = $this->getMockBuilder('Cake\Database\Schema\Table')
+            ->setConstructorArgs([
+                'uploads',
+                [
+                    'photo' => 'string',
+                    'photo_dir' => 'string'
+                ]
+            ])
+            ->getMock();
+
+        $uploadsTable = $this->getMockBuilder('Cake\ORM\Table')
+            ->setConstructorArgs([
+                ['schema' => $uploadsSchema]
+            ])
+            ->getMock();
+
+        $uploadsTable->method('entityClass')->willReturn('Cake\ORM\Entity');
+        $uploadsTable->method('alias')->willReturn('Uploads');
+        $uploadsTable->method('schema')->willReturn($uploadsSchema);
+        $uploadsTable->method('eventManager')->willReturn($eventManager);
+
+        $config = [
+            'photo' => [
+                'dir' => 'photo_dir',
+                'thumbnailSizes' => [
+                    'square' => ['w' => 200, 'h' => 200, 'crop' => true],
+                ],
+                'pathClass' => '\Proffer\Tests\Stubs\TestPath'
+            ],
+        ];
+
+        $Proffer = $this->getMockBuilder('Proffer\Model\Behavior\ProfferBehavior')
+            ->setConstructorArgs([$uploadsTable, $config])
+            ->setMethods(['moveUploadedFile'])
+            ->getMock();
+
+        $Proffer->expects($this->exactly(1))
+            ->method('moveUploadedFile')
+            ->willReturnCallback(function ($source, $destination) {
+                if (!file_exists(pathinfo($destination, PATHINFO_DIRNAME))) {
+                    mkdir(pathinfo($destination, PATHINFO_DIRNAME), 0777, true);
+                }
+
+                return copy($source, $destination);
+            });
+
+        $entity = new Entity([
+            'name' => 'image_640x480.jpg',
+            'tmp_name' => Plugin::path('Proffer') . 'tests' . DS . 'Fixture' . DS . 'image_640x480.jpg',
```

Comment:
Looks like you're using global state here in the test; is `Plugin::path` pure? What can affect it? Can varying return values result in different test results from expectations?